packages/bezier-react/src/components/AlphaToggleEmojiButtonGroup/ToggleEmojiButtonGroup.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { type CSSProperties, forwardRef, useState } from 'react'
+
+import * as ToggleGroup from '@radix-ui/react-toggle-group'
+import classNames from 'classnames'
+
+import useMergeRefs from '~/src/hooks/useMergeRefs'
+import { cssDimension } from '~/src/utils/style'
+
+import {
+  EMOJI_BUTTON_GROUP_GAP,
+  EMOJI_BUTTON_SIZE,
+  useResizeButton,
+} from '~/src/components/AlphaToggleEmojiButtonGroup/useResizeButton'
+import { BaseButton } from '~/src/components/BaseButton'
+import { Emoji } from '~/src/components/Emoji'
+
+import {
+  type ToggleEmojiButtonGroupProps,
+  type ToggleEmojiButtonSourceProps,
+} from './ToggleEmojiButtonGroup.types'
+
+import styles from './ToggleEmojiButtonGroup.module.scss'
+
+const EMOJI_SIZE = 30
+
+/**
+ * Toggle Button that contains `Emoji` component with size fixed to 30.
+ * It should be used with `ToggleEmojiButtonGroup` component.
+ * @example
+ * ```tsx
+ * <ToggleEmojiButtonSource
+ *   content={
+ *     <Emoji
+ *       imageUrl="https://cf.exp.channel.io/asset/emoji/images/80/a.png"
+ *       name="A"
+ *    />
+ *   }
+ * />
+ * ```
+ */
+export const ToggleEmojiButtonSource = forwardRef<
+  HTMLButtonElement,
+  ToggleEmojiButtonSourceProps
+>(function ToggleEmojiButtonSource(
+  { name, variant, className, selected, size = 'm', value, onResize, ...rest },
+  forwardedRef
+) {
+  return (
+    <ToggleGroup.Item
+      value={value}
+      asChild
+    >
+      <BaseButton
+        ref={forwardedRef}
+        className={classNames(
+          styles.ToggleEmojiButtonSource,
+          styles[`size-${size}`],
+          styles[`variant-${variant}`],
+          className
+        )}
+        {...rest}
+      >
+        <div className={classNames(styles.ButtonContent)}>
+          <Emoji
+            name={name}
+            size={`${EMOJI_SIZE}`}
+          />
+        </div>
+      </BaseButton>
+    </ToggleGroup.Item>
+  )
+})
+
+/**
+ * Component for grouping `ToggleEmojiButtonSource`.
+ * @example
+ * ```tsx
+ * <ToggleEmojiButtonGroup
+ *   fillDirection="horizontal"
+ * >
+ *   <ToggleEmojiButtonSource content={<Emoji />} />
+ *   <ToggleEmojiButtonSource content={<Emoji />} />
+ * </ToggleEmojiButtonGroup>
+ * ```
+ */
+export const ToggleEmojiButtonGroup = forwardRef<
+  HTMLDivElement,
+  ToggleEmojiButtonGroupProps
+>(function ToggleEmojiButtonGroup(
+  {
+    fillDirection,
+    value,
+    className,
+    defaultValue,
+    children,
+    style,
+    dir = 'ltr',
+    onValueChange,
+    ...rest
+  },
+  forwardedRef
+) {
+  const [ref, setRef] = useState<null | HTMLDivElement>(null)
```

Comment:
```suggestion
  const [container, setContainer] = useState<null | HTMLDivElement>(null)
```

(마이너) React RefObject가 아니라서 이 네이밍이 더 적절해보여요